src/Uno.UI/UI/Xaml/Controls/Flyout/FlyoutPresenter.cs

```diff
@@ -12,6 +13,24 @@ public partial class FlyoutPresenter : ContentControl
 		public FlyoutPresenter()
 		{
 			DefaultStyleKey = typeof(FlyoutPresenter);
+
+			KeyDown += OnKeyDown;
+		}
+
+		private void OnKeyDown(object sender, KeyRoutedEventArgs args)
+		{
+			if (args.Handled)
+			{
+				return;
+			}
+
+			if (args.Key == VirtualKey.Escape)
+			{
+				if (Parent is FlyoutBasePopupPanel fbpp && fbpp.Popup is { } popup)
+				{
+					popup.AssociatedFlyout.Close();
+				}
+			}
 		}
```

Comment:
Consider overriding [`OnKeyDown`](https://github.com/unoplatform/uno/blob/a25fdc5832f4144d4470f5ee4a2c3c06aa9d0d00/src/Uno.UI/UI/Xaml/Controls/Control/Control.cs#L1015) instead of subscribing to `KeyDown`.